Typeracer_Game/src/typeracer_game/GameStarter.java

```diff
@@ -43,34 +54,118 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
             .addGroup(layout.createSequentialGroup()
-                .addGap(112, 112, 112)
+                .addContainerGap()
+                .addComponent(titleLabel, javax.swing.GroupLayout.PREFERRED_SIZE, 95, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addGap(18, 18, 18)
+                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
+                    .addComponent(nameLabel)
+                    .addComponent(nameField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
+                .addGap(18, 18, 18)
                 .addComponent(startButton)
-                .addContainerGap(169, Short.MAX_VALUE))
+                .addGap(18, 18, 18)
+                .addComponent(quitButton)
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, 42, Short.MAX_VALUE)
+                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING, false)
+                    .addComponent(messageLabel, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
+                    .addComponent(serverButton, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
+                .addContainerGap())
         );
 
         pack();
     }// </editor-fold>//GEN-END:initComponents
 
 private void startButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_startButtonActionPerformed
 // TODO add your handling code here:
-    Game ob = new Game();
-    this.setVisible(false);
-    ob.setVisible(true);
-
+    new Thread () {
+      @Override public void run () {
+      //  GreenhouseControls.startMeUp();
+      }
+    }.start();
+    if(nameField.getText().equals("")){
+        JOptionPane.showMessageDialog(rootPane, "Enter Name");
+    }
+    else{
+        Game ob;
+        try {
+            ob = new Game(nameField.getText());
+            this.setVisible(false);
+            ob.setVisible(true);
+        //    Thread thread = new Thread((Runnable) ob);
+        //    thread.start();
+        } catch (IOException ex) {
+            Logger.getLogger(GameStarter.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
 }//GEN-LAST:event_startButtonActionPerformed
 
+    private void quitButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_quitButtonActionPerformed
+        // TODO add your handling code here:
+        System.exit(0);
+    }//GEN-LAST:event_quitButtonActionPerformed
+
+    private void serverButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_serverButtonActionPerformed
+        // TODO add your handling code here:
+        messageLabel.setText("Server Started and Running ...");
+        messageLabel.paintImmediately(messageLabel.getVisibleRect());
+        GameServer object = new GameServer();
```

Comment:
GameServer class doesn't exist !??

Reply:
sorry i forget to add them. Now i added these class

Reply:
@Rashi-G i have done multiplayer and its working.
i think the best final project is gonna to be merge?

Reply:
whatever, in my code it also solved fix #6

Reply:
Nice! Let me have a look :)